docs/content/_index.md

```diff
@@ -1,608 +1,1007 @@
+true _if_ the rule body is true for some set of variable assignments. In the
+example above `any_public_networks = true` is the head and `net :=
+input.networks[_]; net.public` is the body.
 
-```ruby
-import data.servers                            # Base document
-import data.opa.examples.violations            # Virtual document
+You can query for the value generated by rules just like any other value:
+
+```live:example/complete/1:query
+any_public_networks
+```
+```live:example/complete/1:output
 ```
 
-or as a URI component in an HTTP request:
+All values generated by rules can be queried via the global `data` variable.
 
-```http
-GET https://example.com/v1/data/servers HTTP/1.1
+```live:example/complete/1/abs:query:merge_down
+data.example.rules.any_public_networks
+```
+```live:example/complete/1/abs:output:merge_down
 ```
 
-```http
-GET https://example.com/v1/data/opa/examples/violations HTTP/1.1
+> 💡 You can query the value of any rule loaded into OPA by referring to it with an
+> absolute path. The path of a rule is always:
+> `data.<package-path>.<rule-name>`.
+
+If you omit the `= <value>` part of the rule head the value defaults to `true`.
+You could rewrite the example above as follows without changing the meaning:
+
+```live:example/complete_elided:module:read_only,openable
+package example.rules
+
+any_public_networks {
+    net := input.networks[_]
+    net.public
+}
 ```
 
-> Since the `data` document includes both base and virtual documents,
-> it is possible to query for both at the same time. The easiest way
-> to illustrate this is to query for _all_ of `data` at once. Note,
-> OPA does NOT allow base and virtual documents to overlap. For
-> example, if you try to load a rule that defines a virtual document
-> at path a/b/c (which is already defined by a base document), OPA
-> will return an error. Similarly, if you try to load a base document
-> into a path that is already defined by a virtual document, OPA will
-> also return an error.
+Similarly, if you omit the rule body it defaults to `true`. This allows you to
+define constants. For example:
 
-### The `input` Document
+```live:example/complete_constant:module:openable
+package example.constants
 
-In some cases, policies require input values. In addition to the built-in
-`data` document, OPA also has a built-in `input` document. When you query
-OPA, you can set the value of the `input` document.
+pi = 3.14
+```
 
-Example `input` document:
+Constants defined like this can be queried just like any other values:
 
-```json
+```live:example/complete_constant:query:merge_down
+pi > 3
+```
+```live:example/complete_constant:output
+```
+
+If OPA cannot find variable assignments that satisfy the rule body, we say that
+the rule is undefined. For example, if the `input` provided to OPA did not
+include a public network then `any_public_networks` will be undefined (which is
+not the same as false.)
+
+```live:example/complete/1/undefined:input:merge_down
```

Comment:
Oh.. maybe ignore my comment above about giving a hint to change the input to have no public networks 😄 

It might be useful to make it more clear that this object is the new input being used. I know because I looked at the markdown to confirm, but a new user playing with the live docs maybe wouldn't know that. Maybe this is more of a feature request of the live blocks to have some way to link together the input/query/module/output sections visually so a reader of the docs knows which ones are associated.

Reply:
I've added a note to clarify that the input is different.